plugins/LayerView/layers3d.shader

```diff
@@ -0,0 +1,346 @@
+        gl_Position = u_viewProjectionMatrix * (gl_in[0].gl_Position + g_vertex_offset_horz);
+        EmitVertex();
+
+        f_normal = g_vertex_normal_vert;
+        gl_Position = u_viewProjectionMatrix * (gl_in[0].gl_Position + g_vertex_offset_vert);
+        EmitVertex();
+
+        f_normal = g_vertex_normal_horz_head;
+        gl_Position = u_viewProjectionMatrix * (gl_in[0].gl_Position + g_vertex_offset_horz_head);
+        EmitVertex();
+
+        f_normal = -g_vertex_normal_horz;
+        gl_Position = u_viewProjectionMatrix * (gl_in[0].gl_Position - g_vertex_offset_horz);
+        EmitVertex();
+
+        EndPrimitive();
+
+        f_normal = -g_vertex_normal_horz;
+        gl_Position = u_viewProjectionMatrix * (gl_in[0].gl_Position - g_vertex_offset_horz);
+        EmitVertex();
+
+        f_normal = -g_vertex_normal_vert;
+        gl_Position = u_viewProjectionMatrix * (gl_in[0].gl_Position - g_vertex_offset_vert);
+        EmitVertex();
+
+        f_normal = g_vertex_normal_horz_head;
+        gl_Position = u_viewProjectionMatrix * (gl_in[0].gl_Position + g_vertex_offset_horz_head);
+        EmitVertex();
+
+        f_normal = g_vertex_normal_horz;
+        gl_Position = u_viewProjectionMatrix * (gl_in[0].gl_Position + g_vertex_offset_horz);
+        EmitVertex();
+
+        EndPrimitive();
+
+        // right side
+        f_vertex = v_vertex[1];
+        f_color = v_color[1];
+
+        f_normal = g_vertex_normal_horz;
+        gl_Position = u_viewProjectionMatrix * (gl_in[1].gl_Position + g_vertex_offset_horz);
+        EmitVertex();
+
+        f_normal = g_vertex_normal_vert;
+        gl_Position = u_viewProjectionMatrix * (gl_in[1].gl_Position + g_vertex_offset_vert);
+        EmitVertex();
+
+        f_normal = -g_vertex_normal_horz_head;
+        gl_Position = u_viewProjectionMatrix * (gl_in[1].gl_Position - g_vertex_offset_horz_head);
+        EmitVertex();
+
+        f_normal = -g_vertex_normal_horz;
+        gl_Position = u_viewProjectionMatrix * (gl_in[1].gl_Position - g_vertex_offset_horz);
+        EmitVertex();
+
+        EndPrimitive();
+
+        f_normal = -g_vertex_normal_horz;
+        gl_Position = u_viewProjectionMatrix * (gl_in[1].gl_Position - g_vertex_offset_horz);
+        EmitVertex();
+
+        f_normal = -g_vertex_normal_vert;
+        gl_Position = u_viewProjectionMatrix * (gl_in[1].gl_Position - g_vertex_offset_vert);
+        EmitVertex();
+
+        f_normal = -g_vertex_normal_horz_head;
+        gl_Position = u_viewProjectionMatrix * (gl_in[1].gl_Position - g_vertex_offset_horz_head);
+        EmitVertex();
+
+        f_normal = g_vertex_normal_horz;
+        gl_Position = u_viewProjectionMatrix * (gl_in[1].gl_Position + g_vertex_offset_horz);
+        EmitVertex();
+
+        EndPrimitive();
+    }
+
+fragment41core =
+    #version 410
+    in lowp vec4 f_color;
+    in lowp vec3 f_normal;
+    in lowp vec3 f_vertex;
+
+    out vec4 frag_color;
+
+    uniform mediump vec4 u_ambientColor;
```

Comment:
I think all of our codestyle was like that. Could be that someone ignored it in the past.